src/lib/shim/src/lib.rs

```diff
@@ -433,4 +546,67 @@ pub mod export {
     pub extern "C" fn _shim_load() {
         load();
     }
+
+    /// # Safety
+    ///
+    /// `blk` must contained a serialized block of
+    /// type `ManagerShmem`, which outlives the current process.
+    #[no_mangle]
+    pub unsafe extern "C" fn _shim_set_manager_shmem(shmem: *const ShMemBlockSerialized) {
+        let shmem = unsafe { shmem.as_ref().unwrap() };
+        unsafe { global_manager_shmem::set(shmem) };
+    }
+
+    #[no_mangle]
+    pub extern "C" fn shim_managerSharedMem(
+    ) -> *const shadow_shim_helper_rs::shim_shmem::export::ShimShmemManager {
+        let rv = global_manager_shmem::try_get();
+        rv.map(|x| {
+            let rv: &shadow_shim_helper_rs::shim_shmem::export::ShimShmemManager = &x;
+            rv as *const _
+        })
```

Comment:
This seems a bit confusing since it's returning a reference to something passed by value to the closure (the closure takes an `x` and returns `&x`). I'm guessing that `deref` is being called here somewhere implicitly? Maybe make this more explicit?

Reply:
Right. Added comments and an explicit call to `deref`